### ISSUE TITLE
Redo experiments with Ben's Walker3D as well as HalfCheetah

### ISSUE THREAD

**Issue by farzadab** [farzadab/walking-benchmark#13]
- [x] clean up Friday experiment's code and commit
   - [x] fix: save the reward net?
 - [x] run the 4 experiment with `HalfCheetah`
 - [x] plot the results and discuss
  --
 - [ ] investigate why training is slower
  -- 
 - [ ] get familiar with Ben's `Walker3d`
 - [ ] define the 4 rewards and pass them as args
 - [ ] run the 4 experiments with `Walker3d`
 - [ ] plot and conclude

**Comment by farzadab** [comment]
For the reference, this is the original plot for `Walker2D`:

Experiment IDs:
 - `Apr26_17-12-55_Agility2-baseline`
 - `Apr26_17-13-09_Agility2-shared-value-policy`
 - `Apr26_17-14-12_Agility2-supervised+shared`
 - `Apr26_17-14-55_Agility2-supervised-not-shared`


![image](https://user-images.githubusercontent.com/5062458/56913203-e3f16500-6a65-11e9-8b30-eddcab99a1d0.png)

**Context:** Walker2D with 1.5x torque limit and 0.5 `alive_coef`. Normally the baseline should solve this task but the network size (depth) is too high in this case.

**Discussion:** seems like the supervised (shared reward-policy) version is unstable. Probably needs more time to be fixed, but the shared value-policy seems to be a clear winner for now.

**Modification:** Lowering LR by 10.

**Comment by farzadab** [comment]
### HalfCheetah Experiment Results

Again sharing of value and policy networks seems to be useful. The supervised learning part doesn't seem to have detrimental effect either but maybe that's just the result of using a lower `learning_rate` (`args.vf_lr / 10`).

Ben (@belinghy) also tried this idea for `Walker3D` but **did not get positive results**. Need to investigate the reason. My hypothesis: the interference of the SL becomes too much for harder tasks.

Experiment IDs:
 - `Apr29_10-20-33_Agility2-cheetah-baseline`
 - `Apr29_10-22-37_Agility2-cheetah-shared-value-policy`
 - `Apr29_10-25-12_Agility2-cheetah-supervised+shared`
 - `Apr29_10-25-32_Agility2-cheetah-supervised-not-shared`
 - `Apr29_12-39-00_Agility2-cheetah-baseline-11epoch-longer`

![image](https://user-images.githubusercontent.com/5062458/56934086-909c0880-6a9f-11e9-9ba9-65dc7fdc35b8.png)

**Comment by farzadab** [comment]
**Issue:** the training seems to be quite a bit slower even for the baseline, not sure why.
